resolwe_bio/tools/samplehcluster.py

```diff
@@ -49,18 +49,16 @@
 
 # read data
 matrix = []
+gene_subset = set(args.genes) if args.genes else None
 for fname in args.sample_files:
-    matrix.append([])
-
     with gzip.open(fname) as csvfile:
         sample_reader = csv.reader(csvfile, delimiter=b'\t')
         next(sample_reader)  # Skip the header
 
-        for row in sample_reader:
-            gene_name, gene_value = row
-
-            if not args.genes or gene_name in args.genes:
-                matrix[-1].append(float(gene_value))
+        if gene_subset is not None:
```

Comment:
Write here `if gene_subset is None:` and change the order of the two statements below. It's a bit shorter and more transparent.